app/controllers/api/v1/cinema_halls_controller.rb

```diff
@@ -3,52 +3,47 @@ class Api::V1::CinemaHallsController < ApplicationController
 
   #GET    /api/v1/cinema_hall
   def index
-    @cinema_halls = CinemaHall.all
-
-    render json: @cinema_halls
+    CinemaHalls::UseCases::Index.new.call
   end
 
   def show
-
+    CinemaHalls::UseCases::Show.new.call(id: params[:id])
```

Comment:
In this sample project you can also see how to get around the `.new.call` thing. :) 

```
def self.call(params:)
      new(params: params).call
end
```